test/hashmaps/test_maps.fypp

```diff
@@ -53,10 +54,10 @@ contains
             type(error_type), allocatable, intent(out) :: error
 
             type(chaining_hashmap_type)   :: map
-            integer(int8)             :: test_8_bits(test_size)
+            integer(int8)             :: test_8_bits(test_size,key_types)
 
             call generate_vector(test_8_bits)
```

Comment:
`test_8_bits` is not a vector anymore.
Could you explain why `test_8_bits`  is changed to an array? I am not sure to understand the aim here.

Reply:
The approach is to create a unique int8 vector for each key type that can then be used via transfer or other mechanism to generate a unique key hash for the test program.   With this approach, using a single int8 vector will result in duplicate keys and mapping conflicts, so that must be managed.   There are other approaches, but this seemed to be the easiest additional and extensible to future key types.  For example, we could add a dedicated int32 vector generation function, but not sure how that improves on using transfer with an int8 array to create an int32 array, and would require a dedicated function for each key type.  (So for example if there is want to add int64 or real key types, additional functions would need to be written).

Reply:
Its also worth noting that the structure of test_maps.fypp is not ideal with the code base being repeated twice for open and chaining maps.  I believe refactoring this code into a single source approach should be done before any additional function additions.

Reply:
Hopefully this is straightforward with `fypp`.

Reply:
> Its also worth noting that the structure of test_maps.fypp is not ideal with the code base being repeated for open and chaining maps. I believe refactoring this code into a single source approach should be done before any additional function additions.

Could you open an issue for this, please?